### PR TITLE
Update linking-a-pull-request-to-an-issue.md

### LINKED ISSUE #165: 'Linking an issue to PR' doco should highlight it only works on the default branch
### What feature or product is affected?

The `managing-your-work-on-github/linking-a-pull-request-to-an-issue.md` page which covers how special keywords can be used in pull request descriptions. 

Basically, the code looks in the description for special terms and parses them to create links to issues, and additionally, issues are closed when the pull request is merged.

The special keywords in a pull request description are interpreted when the pull request targets the repository's _default_ branch.

However, if the PR's base is *any other branch*, then these keywords are ignored, no links are created and merging the PR has no effect on the issues.

### What is the new or expected behavior?

In particular, we should mention that both linking and closing behaviours *only* apply to PRs that target the repository's default branch.

Secondly, IMHO this condition is important enough that we should also highlight it, perhaps in a big yellow box or similar. Having it in a single sentence as part of a paragraph makes it much too easy to miss.

### How is the old or inaccurate behavior currently documented?

In `managing-your-work-on-github/linking-a-pull-request-to-an-issue.md`, the first section mentions that issues can be linked from a pull request. However, this is misleading and incomplete, as it does not say that this only happens when the PR base branch is the repository default branch.

The closing section does allude to the behaviour when it says:

> When you merge a linked pull request into the default branch of a repository, its linked issue is automatically closed.

This is a single statement, so it is very easy to miss amongst the rest of the document and our documentation.

It is also very weakly stated - it only applies to one half the situation. This is not just an "if" condition, it is an "if and only if". 

### Content to update

In `managing-your-work-on-github/linking-a-pull-request-to-an-issue.md`:

- [x] This doc does mention the default branch passively a number of times, but we should be more explicit in the fact that it only works on the default branch.
- [x] Add a note or short paragraph at the top of "Linking a pull request to an issue using a keyword" that it must be on the default branch 
- [x] While we're in the article, lets update a line in [Linking a pull request to an issue](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) change the sentence to "You can link a pull request to an issue by using a supported keyword in the pull request's description or in a commit message."

### PR DESCRIPTION
**Pull request by casals**
### Why:

Fixes #165 .

### What's being changed:

I added a Note box at the top of the document, and a reference to it at the top of the document - should be enough warning without disrupting the rest of the text. Also changed the introductory line in the "Linking a pull request(...)" section as suggested by the author/PR reviewer:

- Original sentence: "You can link a pull request to an issue by using a supported keyword in the pull request's description."
- Modified sentence: "You can link a pull request to an issue by using a supported keyword in the pull request's description or in a commit message (please note that the PR must be on the default branch)."

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
Amazing @casals! I'll send this one to @github/docs-content-core team to review, too 👍

**Comment by janiceilene** [comment]
Thanks @casals! I'll let the @github/docs-content-core team know this is ready for review 🌟

**Review by atlslscsrv-app — APPROVED** [review]
@atlslscsrv-app

**Review by felicitymay — COMMENTED** [review]
Hi - thanks for addressing this issue, it's great to have this topic improved ✨ 

I'm a little puzzled by:

> I added a Note box at the end of the document, and a reference to it at the top of the document

On [staging](https://docs-199--patch-10.herokuapp.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), the note box appears to be at the start of the topic. Given that our search links to each H3 within the topic, I'm wondering if the note would be better placed within the "[Linking a pull request to an issue using a keyword](https://docs-199--patch-10.herokuapp.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)" section, maybe straight after the bulleted list of special terms. Otherwise people who go straight to that section will miss the note. What do you think?

If we did move the note box, then probably we should expand the first sentence of the topic: *"You can link an issue to a pull request manually or using a supported keyword in the pull request description."* to mention the default branch.

**Comment by casals** [comment]
Hey @felicitymay - what happened here was: I had just submitted a number of PRs that involved including a Note box at the end of the document. During the review of the first of these PRs, one of the requested changes was to move the Note box to the top of the page, since that's the standard style/practice. In order to maintain consistency, I moved the (new) Note box to the top of the document in all of my PRs - but forgot to change the description here, sorry.

All that being said, I think we can go with your latter suggestion: keep the box at the topic of the document + expand the first sentence of the topic. In fact, this sentence was already extended from its original version. I'm updating the PR description with these changes.

File: content/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue.md [hunk 1]
```diff
@@ -37,7 +43,7 @@ You can manually link up to ten issues to each pull request. The issue and pull
 
 ### Linking a pull request to an issue using a keyword
 
-You can link a pull request to an issue by using a supported keyword in the pull request's description.
+You can link a pull request to an issue by using a supported keyword in the pull request's description or in a commit message (please note that the PR must be on the default branch).
```
Comment (felicitymay) [inline content/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue.md:46]: This looks like a good solution 😄 Thanks 🙇🏻‍♀️ 

We generally write out "pull request" in full (although I can see a few topics do use PR).

```suggestion
You can link a pull request to an issue by using a supported keyword in the pull request's description or in a commit message (please note that the pull request must be on the default branch). 
```

**Review by felicitymay — APPROVED** [review]
Thanks 👍🏻 🚀

**Comment by felicitymay** [comment]
Thanks for responding so quickly - it's much appreciated. I've added the `automerge` label and will check back to make sure that all's well.

**Comment by felicitymay** [comment]
Ah. Okay. Will check back to see when the tests have passed and merge.

**Pull request merged**